### ISSUE TITLE
E1010 GetAtt check fails on AWS::SageMaker::CodeRepository resource 

### ISSUE THREAD

**Issue by seporaitis** [aws-cloudformation/cfn-python-lint#1130]
*cfn-lint version*: `0.24.1`

*Description of issue*

`GetAtt` (E1010) lint rule fails with false positive saying an instance of `AWS::SageMaker::CodeRepository` does not have an attribute `CodeRepositoryName`. However, the [official resource documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sagemaker-coderepository.html#aws-resource-sagemaker-coderepository-return-values) indicates otherwise. Presumably this is an issue with the CloudSpecs. However, `cfn-lint -u` did not resolve the issue.

Minimal template to replicate the issue:

```lang=yaml
AWSTemplateFormatVersion: '2010-09-09'

Parameters:

  SecurityGroup:
    Type: String

  SubnetId:
    Type: String

  RoleArn:
    Type: String

Resources:

  NotebookRepositoryLink:
    Type: AWS::SageMaker::CodeRepository
    Properties:
      GitConfig:
        RepositoryUrl: https://example.com/repo/

  Notebook:
    Type: AWS::SageMaker::NotebookInstance
    Properties:
      DefaultCodeRepository: !GetAtt NotebookRepositoryLink.CodeRepositoryName
      DirectInternetAccess: Disabled
      InstanceType: ml.t3.xlarge
      RootAccess: Disabled
      RoleArn: !Ref RoleArn
      SecurityGroupIds:
      - !Ref SecurityGroup
      SubnetId: !Ref SubnetId
```

Running `cfn-lint test_template.yml` yields:

```
E1010 Invalid GetAtt NotebookRepositoryLink.CodeRepositoryName for resource Notebook
test_template.yml:25:7
```

Expected output - no linter errors.

Temporary workaround is to disable this rule on the resource.

**Comment by PatMyron** [comment]
[`AWS::SageMaker::CodeRepository`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sagemaker-coderepository.html) is now in the official [resource specification for `us-east-1`](https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json):

```json
    "AWS::SageMaker::CodeRepository": {
      "Attributes": {
        "CodeRepositoryName": {
          "PrimitiveType": "String"
        }
      },
```
